packages/web-app-files/src/helpers/resource/move.ts

```diff
@@ -0,0 +1,166 @@
+  hideModal
+) => {
+  const targetFolderItems = await client.files.list(targetFolder.webDavPath, 50)
+  const targetPath = targetFolder.path
+  const index = targetFolder.webDavPath.lastIndexOf(targetPath)
+  const webDavPrefix = targetFolder.webDavPath.substring(0, index)
+
+  // Collect all conflicting resources
+  const allConflicts = []
+  for (const resource of resourcesToMove) {
+    const potentialTargetWebDavPath = join(webDavPrefix, targetFolder.path, resource.path)
+    const exists = targetFolderItems.some((e) => e.name === potentialTargetWebDavPath)
+    if (exists) {
+      allConflicts.push({
+        resource,
+        strategy: null
+      } as FileConflict)
+    }
+  }
+  let count = 0
+  let doForAllConflicts = false
+  let doForAllConflictsStrategy = null
+  const resolvedConflicts = []
+  for (const conflict of allConflicts) {
+    // Resolve conflicts accordingly
+    if (doForAllConflicts) {
+      conflict.strategy = doForAllConflictsStrategy
+      resolvedConflicts.push(conflict)
+      continue
+    }
+
+    // Resolve next conflict
+    const conflictsLeft = allConflicts.length - count
+    const result: ResolveConflict = await resolveFileExists(
+      createModal,
+      hideModal,
+      conflict.resource.name,
+      conflictsLeft
+    )
+    conflict.strategy = result.strategy
+    resolvedConflicts.push(conflict)
+    count += 1
+
+    // User checked 'do for all x conflicts'
+    if (!result.doForAllConflicts) continue
+    doForAllConflicts = true
+    doForAllConflictsStrategy = result.strategy
+  }
+  return resolvedConflicts
+}
+const showResultMessage = async(errors, movedResources, showMessage) => {
+  if (errors.length === 0) {
+    const count = movedResources.length
+    const title = `${count} item was moved successfully`
+    showMessage({
+      title,
+      status: 'success'
+    })
+    return 
+  }
+
+  if (errors.length === 1) {
+    const title = `Failed to move "${errors[0]?.resourceName}`
+    showMessage({
+      title,
+      status: 'danger'
+    })
+    return
+  }
+
+  const title = `Failed to move ${errors.length} resources`
+  showMessage({
+    title,
+    status: 'danger'
+  })
+}
+export const move = async (
+  resourcesToMove,
+  targetFolder,
+  client,
+  createModal,
+  hideModal,
+  showMessage
+) => {
+  const errors = []
+  const resolvedConflicts = await resolveAllConflicts(
+    resourcesToMove,
+    targetFolder,
+    client,
+    createModal,
+    hideModal
+  )
+  const movedResources = []
+
+  for (const resource of resourcesToMove) {
+    const hasConflict = resolvedConflicts.some((e) => e.resource.id === resource.id)
+    if (hasConflict) {
+      const resolveStrategy = resolvedConflicts.find((e) => e.resource.id === resource.id)?.strategy
+      if (resolveStrategy === ResolveStrategy.CANCEL) {
+        continue
+      }
+      if (resolveStrategy === ResolveStrategy.REPLACE) {
+        await client.files.delete(join(targetFolder.webDavPath, resource.name))
```

Comment:
Do we need to delete via client code? I'd expect this to be handled by the backend 🤔 Not sure how it's implement. If we need to handle it via client code then it might be more safe to first rename the file, then move the other one and lastly delete the previously renamed file.

Reply:
I've seen an overwrite flag for restoring files in the sdk, not sure if its possible with move, have looked it up and didn't find a way

Reply:
ok fixed it